lib/src/slices/data_bkg/data_bkg_service.dart

```diff
@@ -64,8 +64,24 @@ class DataBkgService extends ChangeNotifier {
                 .subtract(Duration(days: 1))
                 .isAfter(gmailLastFetch))) {
       DateTime run = DateTime.now();
-      await _checkGmailFetchList(
-          fetchAll: fetchAll, lastChecked: gmailLastFetch);
+      ApiAppDataModel? savedActiveCategory =
+          await _apiAppDataService.getByKey(ApiAppDataKey.gmailCategory);
+      String activeCategory = savedActiveCategory?.value ?? '';
+      int categoryStartIndex = activeCategory.isEmpty
+          ? 0
+          : model.gmailCategoryList.indexOf(activeCategory);
+      for (int i = categoryStartIndex;
+          i < model.gmailCategoryList.length;
+          i++) {
+        String category = model.gmailCategoryList[i];
+        await _apiAppDataService.save(ApiAppDataKey.gmailCategory, category);
```

Comment:
if you move the save to after the fetch you can use a ternary save and include line 83 at the same time, also saves 1 duplicated write to the DB since the first iteration of this loop will re-save the value fetched in 67.